core/src/test/java/org/infinispan/test/MultipleCacheManagersTest.java

```diff
@@ -420,8 +425,26 @@ protected DataContainer dataContainer(int index) {
       return advancedCache(index).getDataContainer();
    }
 
-   @Factory
+   /**
+    * This is the method you should override when providing factory method.
+    */
    public Object[] factory() {
+      throw new IllegalStateException("Only overridden methods should be called!");
+   }
+
+   @Factory
+   public Object[] defaultFactory() {
+      // It is possible to override the factory method, but if we extend a class that defines such overridden
+      // method, the factory method will be inherited, too - that results in running the superclass tests
```

Comment:
Please, comment on this rather in https://github.com/infinispan/infinispan/pull/4487.
I don't like it either. I don't like TestNG anymore :) Do you have a better solution for this problem? (I did not understand your suggestion).